src/JsonSchema/Constraints/ConstraintInterface.php

```diff
@@ -59,5 +59,10 @@ public function isValid();
      * @param mixed            $i
      * @throws \JsonSchema\Exception\ExceptionInterface
      */
+    public function _check(&$value, $schema = null, JsonPointer $path = null, $i = null);
```

Comment:
OK - I'm certainly new to this project, and if you guys are OK with breaking the API, then that's your call, and I'll go along with that.

So... I will make it so that this PR has one top-level method named `check`, which will do what `validate` does now. All the other ones will go. Does that sound OK?

`_check` will need to stay, unless I do some significant refactoring to change the way those calls work. Keeping it there seems like the most sensible option at this point - however if you feel strongly about that, in prepared to do that work.